onchain/rollups/contracts/library/LibInput.sol

```diff
@@ -12,36 +12,57 @@ library LibInput {
     /// @notice Raised when input is larger than the machine limit.
     error InputSizeExceedsLimit();
 
-    /// @notice Summarize input data in a single hash.
+    /// @notice Encode an EVM input
     /// @param sender `msg.sender`
     /// @param blockNumber `block.number`
     /// @param blockTimestamp `block.timestamp`
-    /// @param input The input blob
-    /// @param inputIndex The index of the input in the input box
-    /// @return The input hash
-    function computeInputHash(
+    /// @param index The index of the input in the input box
+    /// @param payload The input payload
+    /// @return The encoded EVM input
+    function encodeEvmInput(
         address sender,
         uint256 blockNumber,
         uint256 blockTimestamp,
-        bytes calldata input,
-        uint256 inputIndex
-    ) internal pure returns (bytes32) {
-        if (input.length > CanonicalMachine.INPUT_MAX_SIZE) {
-            revert InputSizeExceedsLimit();
-        }
-
-        bytes32 keccakMetadata = keccak256(
-            abi.encode(
+        uint256 index,
+        bytes calldata payload
+    ) internal pure returns (bytes memory) {
+        return
+            abi.encodeWithSignature(
```

Comment:
So, recently I found a nice way to encode function calls in a nice type-safe way. See my comment on the output unification issue [here](https://github.com/cartesi/rollups-contracts/issues/42#issuecomment-1791371612). I was thinking if we could create an `Inputs.sol` interface on the `contracts/common` directory and define the `EvmInput` function there. What do you think?

This should also simplify the tests, since they will be able to more easily encode EVM inputs with `abi.encodeCall` as well. No need for an auxiliary function that calls the library function, etc.

As a result, we can encode the input directly on the `InputBox` contract. We can also check there if the input is smaller than the maximum input size, and hash it. With these changes, we could delete the `LibInput.sol` file altogether.

Reply:
Good idea Gui 👍 

>  we can encode the input directly on the InputBox contract

Yeah the `computeEvmInputHash` library function is quite short that we can directly write it in the `InputBox` contract.